sale_triple_discount/models/sale_order_line.py

```diff
@@ -131,3 +131,95 @@ def _convert_to_tax_base_line_dict(self):
+                if len(line.tax_id.filtered(lambda tax: tax.price_include)) > 0:
+                    # As included taxes are not excluded from the computed subtotal, `compute_all()` method
+                    # has to be called to retrieve the subtotal without them.
+                    # `price_reduce_taxexcl` cannot be used as it is computed from `price_subtotal` field. (see upper Note)
+                    price_subtotal = line.tax_id.compute_all(
+                        price_reduce,
+                        currency=line.currency_id,
+                        quantity=uom_qty_to_consider,
+                        product=line.product_id,
+                        partner=line.order_id.partner_shipping_id,
+                    )["total_excluded"]
+                inv_lines = line._get_invoice_lines()
+                if any(
+                    inv_lines.mapped(
+                        lambda inv_l: inv_l.discount != line.discount
+                        or inv_l.discount2 != line.discount2
+                        or inv_l.discount3 != line.discount3
+                    )
+                ):
+                    # In case of re-invoicing with different discount we try to calculate manually the
+                    # remaining amount to invoice
+                    amount = 0
+                    for inv_line in inv_lines:
+                        if (
+                            len(
+                                inv_line.tax_ids.filtered(lambda tax: tax.price_include)
+                            )
+                            > 0
+                        ):
+                            amount += inv_line.tax_ids.compute_all(
+                                inv_line.currency_id._convert(
+                                    inv_line.price_unit,
+                                    line.currency_id,
+                                    line.company_id,
+                                    inv_line.date or fields.Date.today(),
+                                    round=False,
+                                )
+                                * inv_line.quantity
+                            )["total_excluded"]
+                        else:
+                            amount += (
+                                inv_line.currency_id._convert(
+                                    inv_line.price_unit,
+                                    line.currency_id,
+                                    line.company_id,
+                                    inv_line.date or fields.Date.today(),
+                                    round=False,
+                                )
+                                * inv_line.quantity
+                            )
+
+                    amount_to_invoice = max(price_subtotal - amount, 0)
+                else:
+                    amount_to_invoice = price_subtotal - line.untaxed_amount_invoiced
+
+            line.untaxed_amount_to_invoice = amount_to_invoice
```

Comment:
Here you are copying a fairly big amount of code from the Odoo function `_compute_untaxed_amount_to_invoice`.
Can you please state this in a comment and use another comment to highlight what are you changing from this function?

Reply:
hi @grindtildeath I want to calculate the 'price_reduce' based on three discount fields. However, the original function doesn't have any separate functions for calculating each discount individually.